model/host/host.go

```diff
@@ -386,7 +386,7 @@ func (h *Host) SetAgentRevision(agentRevision string) error {
 }
 
 // SetNeedsNewAgent sets the "needs new agent" flag on the host
-func (h *Host) SetNeedsNewAgent() error {
+func (h *Host) SetNeedsNewAgent(needsAgent bool) error {
```

Comment:
The implementation doesn't seem to use this flag.

I'd also make "SetNeedsNewAgent" and "UnsetNeedsNewAgent" methods

Reply:
The other changes in the APIs/taskrunner use this added parameter. I like it better this way since it prevents code duplication/unnecessary wrappers that don't do much